beater/beater.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 type beater struct {
-	done   chan struct{}
+	errCh  chan error
```

Comment:
> I'm hesitant to call this errCh as the show down is not only an erro
Not true, it is always an error: https://github.com/golang/go/blob/d6cada7fa4ef92190c2b59c9048413f9280e4a74/src/net/http/server.go#L2680

I found `errCh` as variable name in golang source code itself, surely is idiomatic enough.

> there are potentially 2 different channels needed
I'm pretty sure that 2 channels are not required to shutdown a server.